### PR TITLE
Use get direct in KV

### PR DESCRIPTION
**Pull request by Jarema**

### CODE REVIEW

**Comment by Jarema** [comment]
implements https://github.com/nats-io/nats.rs/pull/651

File: async-nats/src/jetstream/stream.rs [hunk 1]
```diff
@@ -347,11 +347,16 @@ impl Stream {
                 context: self.context.clone(),
                 message,
             })?;
+        println!("RESPOSNE: {:?}", response);
```
Comment (caspervonb) [inline async-nats/src/jetstream/stream.rs:350]: Take out printlns, use trace or debug?

Comment (Jarema) [reply async-nats/src/jetstream/stream.rs]: cleanud up those.

File: async-nats/src/jetstream/response.rs [hunk 2]
```diff
@@ -51,3 +51,20 @@ pub enum Response<T> {
     Err { error: Error },
     Ok(T),
 }
+
+#[derive(Debug)]
+pub struct DirectError {
```
Comment (caspervonb) [inline async-nats/src/jetstream/response.rs:56]: Hold off on specializing this error for the time being? We'll do all of them together instead.

**Review by caspervonb — CHANGES_REQUESTED** [review]

File: async-nats/src/jetstream/kv/mod.rs [hunk 3]
```diff
@@ -217,41 +217,137 @@ impl Store {
...
+                            operation,
+                            raw_message.sequence,
+                            raw_message.time,
+                        ))
+                    }
+                    Err(err) => {
+                        let e: std::io::Error = *err.downcast().unwrap();
+                        let d = e.get_ref().unwrap();
+                        let de = d.downcast_ref::<response::Error>().unwrap();
+                        if de.code == 10037 {
```
Comment (caspervonb) [inline async-nats/src/jetstream/kv/mod.rs:322]:
```suggestion
                        // Doc me what this error is
                        if de.code == 10037 {
```

Comment (Jarema) [reply async-nats/src/jetstream/kv/mod.rs:321]: fixed.

**Review by caspervonb — APPROVED** [review]
lgtm

**Pull request merged**